ui/src/app/components/rollouts-table/rollouts-table.tsx

```diff
@@ -0,0 +1,272 @@
+import {InfoItemKind, InfoItemRow} from '../info-item/info-item';
+import './rollouts-table.scss';
+
+export const RolloutsTable = ({
+    rollouts,
+    onFavoriteChange,
+    favorites,
+}: {
+    rollouts: RolloutInfo[];
+    onFavoriteChange: (rolloutName: string, isFavorite: boolean) => void;
+    favorites: {[key: string]: boolean};
+}) => {
+    const tableRef = React.useRef(null);
+
+    const handleFavoriteChange = (rolloutName: string, isFavorite: boolean) => {
+        onFavoriteChange(rolloutName, isFavorite);
+    };
+    const data = rollouts
+        .map((rollout) => {
+            return {
+                ...rollout,
+                key: rollout.objectMeta?.uid,
+                favorite: favorites[rollout.objectMeta?.name] || false,
+            };
+        })
+        .sort((a, b) => {
+            if (a.favorite && !b.favorite) {
+                return -1;
+            } else if (!a.favorite && b.favorite) {
+                return 1;
+            } else {
+                return 0;
+            }
+        });
+
+    const columns = [
+        {
+            dataIndex: 'favorite',
+            key: 'favorite',
+            render: (favorite: boolean, rollout: RolloutInfo) => {
+                return favorite ? (
+                    <button
+                        onClick={(e) => {
+                            e.stopPropagation();
+                            handleFavoriteChange(rollout.objectMeta?.name, false);
+                        }}
+                        style={{cursor: 'pointer'}}
+                    >
+                        <FontAwesomeIcon icon={faStarSolid} size='lg' />
+                    </button>
+                ) : (
+                    <button
+                        onClick={(e) => {
+                            e.stopPropagation();
+                            handleFavoriteChange(rollout.objectMeta?.name, true);
+                        }}
+                        style={{cursor: 'pointer'}}
+                    >
+                        <FontAwesomeIcon icon={faStarOutline as IconDefinition} size='lg' />
+                    </button>
+                );
+            },
+            width: 50,
+        },
+        {
+            title: 'Name',
+            dataIndex: 'objectMeta',
+            key: 'name',
+            width: 300,
+            render: (objectMeta: {name?: string}) => objectMeta.name,
+            sorter: (a: any, b: any) => a.objectMeta.name.localeCompare(b.objectMeta.name),
+        },
+        {
+            title: 'Strategy',
+            dataIndex: 'strategy',
+            key: 'strategy',
+            align: 'left' as const,
+            sorter: (a: any, b: any) => a.strategy.localeCompare(b.strategy),
+            render: (strategy: string) => {
+                return (
+                    <InfoItemRow label={''} items={{content: strategy, icon: strategy === 'BlueGreen' ? 'fa-palette' : 'fa-dove', kind: strategy.toLowerCase() as InfoItemKind}} />
```

Comment:
add css tyle to make it 
<img width="1703" alt="Screenshot 2023-11-09 at 9 42 31 PM" src="https://github.com/argoproj/argo-rollouts/assets/11219262/f11fd66b-8434-4573-b835-bbc107d200a1">

Reply:
I wasn't able to get this to work with center alignment, but I think it looks really clean when aligned left

Reply:
![image](https://github.com/argoproj/argo-rollouts/assets/35117115/284045d4-597c-4df3-9539-47ca636beb92)